File: frontend-app/src/types/comment.ts
import type { FileType } from "./common";

export interface Comment {
  id: string;
  file_type: FileType;
  file_id: string;
  user_id: string;
  content: string;
  position_json: Record<string, unknown> | null;
  parent_comment_id: string | null;
  resolved: boolean;
  resolved_by: string | null;
  resolved_at: string | null;
  created_at: string;
  updated_at: string;
  author_name: string | null;
  author_avatar: string | null;
}

File: backend/app/schemas/comment.py
from __future__ import annotations
import uuid
from datetime import datetime

from pydantic import BaseModel, Field


class CommentCreate(BaseModel):
    file_type: str = Field(pattern="^(document|spreadsheet|presentation)$")
    file_id: uuid.UUID
    content: str = Field(min_length=1, max_length=5000)
    position_json: dict | None = None
    parent_comment_id: uuid.UUID | None = None


class CommentUpdate(BaseModel):
    content: str = Field(min_length=1, max_length=5000)


class CommentResponse(BaseModel):
    id: uuid.UUID
    file_type: str
    file_id: uuid.UUID
    user_id: uuid.UUID
    content: str
    position_json: dict | None = None
    parent_comment_id: uuid.UUID | None = None
    resolved: bool
    resolved_by: uuid.UUID | None = None
    resolved_at: datetime | None = None
    created_at: datetime
    updated_at: datetime
    # Populated in API
    author_name: str | None = None
    author_avatar: str | None = None

    model_config = {"from_attributes": True}
